app/controllers/tournaments_controller.rb

```diff
@@ -83,16 +83,22 @@ def save_pools
     @tournament = Tournament.includes(:pools, { pools: :pool_fighters }).find(params[:id])
     success = true
     message = "Pools saved"
-
     params[:pools].each do |key, pool|
       current_pool = @tournament.pools.find(pool[:id])
-      current_pool.pool_fighters.delete_all
-      current_fighters = []
-      pool[:fighters].try(:each) do |fighter|
-        current_fighters << { fighter_id: fighter }
+      current_pool_fighters = current_pool.fighters.pluck(:id) || []
+      new_pool_fighters = pool[:fighters] || []
+
+      current_pool_fighters.each do |fighter_id|
```

Comment:
You should move this logic to the `Pool` model, so you can just call that method on `current_pool`.